### DISCUSSION TITLE
Getting the fully qualified database table name from upstream dbt asset

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [dagster-io/dagster#25175] [Q&A]
If I have a non-dbt asset (A) that depends on a dbt asset (B), how can I obtain the full name of the database artifact (i.e. the relation), so I can use it a SQL query called by B.

**Comment by maximearmstrong** [comment]
Hey @the4thamigo-uk - the fully qualified name of a dbt model is included in its materialization metadata with the key “dagster/table_name”. You can access it in a downstream asset using the following code:

```python
from dagster import AssetExecutionContext, AssetKey, asset

@asset
def my_downstream_asset(context: AssetExecutionContext, my_upstream_dbt_asset):
    event = context.instance.get_latest_materialization_event(AssetKey("my_upstream_dbt_asset"))
    table_name = event.asset_materialization.metadata["dagster/table_name"].value
```
